Dockerfile

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright 2020 (C) Olliver Schinagl <oliver@schinagl.nl>
+
+# hadolint ignore=DL3007  latest is the latest stable for alpine
+FROM registry.hub.docker.com/library/alpine:latest AS builder
+
+LABEL Maintainer="Olliver Schinagl <oliver@schinagl.nl>"
+
+WORKDIR /src
+
+COPY . /src/
+
+# hadolint ignore=DL3008  We want the latest stable versions
+RUN apk add --no-cache \
+        bsd-compat-headers \
+        bzip2-dev \
+        cmake \
+        curl-dev \
+        file \
+        fts-dev \
+        g++ \
+        git \
+        json-c-dev \
+        libmilter-dev \
+        libtool \
+        libxml2-dev \
+        linux-headers \
+        make \
+        ncurses-dev \
+        openssl-dev \
+        pcre2-dev \
+        python3 \
+        zlib-dev \
+    && \
+    mkdir -p "./build" && \
```

Comment:
I think we should `python3 -m pip install pytest`.  Our new CTest test suite works with Python's built-in `unittest`, but will be far easier to read test failures in the near future when I merge a PR that supports `pytest`.  It will detect if `pytest` exists and will switch to that if available.

Reply:
you say 'when merged' does that mean it does not work yet? I can add pytest no problem, but if I have no way to verify/validate this. I can use the pytest branch (which one) as a basis as well for the time being ...

Reply:
So coming back to this one, I get one test failure, related to the bytecode stuff? in test 1.
```
1: 99%: Checks: 1112, Failures: 4, Errors: 0
1: /src/unit_tests/check_clamav.c:1460:F:assorted functions:test_cli_codepage_to_utf8_jis:0: test_cli_codepage_to_utf8: Failed to convert CODEPAGE_JAPANESE_SHIFT_JIS to UTF8: ret != SUCCESS!
1: /src/unit_tests/check_clamav.c:1478:F:assorted functions:test_cli_codepage_to_utf8_utf16be_null_term:0: test_cli_codepage_to_utf8: Failed to convert CODEPAGE_UTF16_BE to UTF8: ret != SUCCESS!
1: /src/unit_tests/check_clamav.c:1496:F:assorted functions:test_cli_codepage_to_utf8_utf16be_no_null_term:0: test_cli_codepage_to_utf8: Failed to convert CODEPAGE_UTF16_BE to UTF8: ret != SUCCESS!
1: /src/unit_tests/check_clamav.c:1514:F:assorted functions:test_cli_codepage_to_utf8_utf16le:0: test_cli_codepage_to_utf8: Failed to convert CODEPAGE_UTF16_LE to UTF8: ret != SUCCESS!
```

Here's the link from dockerhub, https://hub.docker.com/repository/docker/olliver/clamav/builds/9fd6e5e1-64f1-4015-9db9-f844ca30f9ec which as you can see 'continues' with the failure. I left that to see if it works (it does) so the goal is to resolve the bug and make it work eventually of course, but I'd need some help there :)

Reply:
`ctest` will work fine without `pytest` right now.  This is a mirror of my internal PR which is being reviewed by a teammate: https://github.com/Cisco-Talos/clamav-devel/pull/153/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR197  In addition to fixing an unrelated issue, it adds support to detect `pytest` and use it instead of `unittest` so the test output is prettier.   If you add `pytest` to the builder layer, it will help later when that has merged.

Your build link redirects to https://hub.docker.com/r/olliver/clamav for me, and doesn't have an option to review the build logs.  I guess they're not public. 

Alpine compatibility is worse than I remembered.  I just reviewed our test logs and those same 4 tests fail for us as well on Alpine.  Testing on Alpine is _relatively_ new for us, and we're ignoring the failures for now.  In addition, we have this line to ignore a test failure on Alpine for one of our QA feature tests that converts some Shift_JIS comments in an EGG archive to utf-8: `@pytest.mark.xfail("musl" in os.popen('ldd /bin/ls').read(), reason="alpine helgrind korean egg test fails for reasons unknown")`